InfiniteLoopBookProject/src/com/st/lms/dao/DAOBookImpl.java

```diff
@@ -0,0 +1,245 @@
+				buffWriter.append(bookLine);
+			}
+			else {
+				bookLine = bookLine.replace(parts[ID], "" + newID);
+				buffWriter.append(bookLine);
+			}
+		}
+		buffReader.close();
+		buffWriter.close();
+		File oldFile = new File(BookFile);
+		oldFile.delete();
+		File newFile = new File(BookTemp);
+		newFile.renameTo(oldFile);
+	}
+
+	@Override
+	public void delete(int id) throws IOException{
+		FileInputStream fin = new FileInputStream(BookFile);
+		FileWriter fw = new FileWriter(BookTemp, true);
+		BufferedWriter bw = new BufferedWriter(fw);
+		PrintWriter fout = new PrintWriter(bw);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String bookLine;
+		while((bookLine = buffReader.readLine()) != null) {
+			String[] parts = bookLine.split(",");
+			if(Integer.parseInt(parts[ID]) != id) {
+				fout.println(bookLine);
+			}
+		}
+		buffReader.close();
+		fout.close();
+		File oldFile = new File(BookFile);
+		oldFile.delete();
+		File newFile = new File(BookTemp);
+		newFile.renameTo(oldFile);
+	}
+	
+	@Override
+	public boolean hasBook(int id) throws IOException{
+		FileInputStream fin = new FileInputStream(BookFile);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String bookLine;
+		
+		while((bookLine = buffReader.readLine()) != null) {
+			String[] parts = bookLine.split(",");
+			if(Integer.parseInt(parts[ID]) == id) {
+				buffReader.close();
+				return true;
+			}
+		}
+		buffReader.close();
+		return false;
+	}
+
+	@Override
+	public void removePublisher(int publisherID) throws IOException {
+		FileInputStream fin = new FileInputStream(BookFile);
+		FileWriter fw = new FileWriter(BookTemp, true);
+		BufferedWriter bw = new BufferedWriter(fw);
+		PrintWriter fout = new PrintWriter(bw);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String bookLine;
+		while((bookLine = buffReader.readLine()) != null) {
+			String[] parts = bookLine.split(",");
+			if(Integer.parseInt(parts[PUBLISHER]) != publisherID) {
+				fout.println(bookLine);
+			}
+		}
+		buffReader.close();
+		fout.close();
+		File oldFile = new File(BookFile);
+		oldFile.delete();
+		File newFile = new File(BookTemp);
+		newFile.renameTo(oldFile);
+	}
+
+	@Override
+	public void removeAuthor(int authorID) throws IOException {
+		FileInputStream fin = new FileInputStream(BookFile);
+		FileWriter fw = new FileWriter(BookTemp, true);
+		BufferedWriter bw = new BufferedWriter(fw);
+		PrintWriter fout = new PrintWriter(bw);
+		BufferedReader buffReader = new BufferedReader(new InputStreamReader(fin));
+		String bookLine;
+		while((bookLine = buffReader.readLine()) != null) {
+			String[] parts = bookLine.split(",");
+			if(Integer.parseInt(parts[AUTHOR]) != authorID) {
+				fout.println(bookLine);
+			}
+		}
+		buffReader.close();
+		fout.close();
+		File oldFile = new File(BookFile);
+		oldFile.delete();
+		File newFile = new File(BookTemp);
+		newFile.renameTo(oldFile);
+	}
+	

```

Comment:
Is there a reason you have so much space between this spot and the end of this class?